packages/api/src/external/fhir/document/index.ts

```diff
@@ -144,3 +156,110 @@ export function idToFHIR(id: DocumentIdentifier): Identifier {
     use: id.use === "unspecified" ? undefined : id.use,
   };
 }
+
+/**
+ * Takes a CW resource and converts it to a FHIR resource
+ *
+ * @param resource
+ * @returns
+ */
+function convertToFHIRResource(resource: Contained): Resource | undefined {
+  if (resource.resourceType === ResourceType.Patient && resource.id) {
+    return {
+      resourceType: ResourceType.Patient,
+      id: resource.id,
+    };
+  } else if (resource.resourceType === ResourceType.Organization && resource.name) {
+    return {
+      resourceType: ResourceType.Organization,
+      name: convertCWNameToString(resource.name),
+    };
+  } else if (resource.resourceType === ResourceType.Practitioner && resource.name) {
+    return {
+      resourceType: ResourceType.Practitioner,
+      name: convertCWNameToHumanName(resource.name),
+    };
+  }
+}
+
+/**
+ * Converts a CW name to a string
+ *
+ * @param name
+ * @returns
+ */
+function convertCWNameToString(name: string | CWHumanName | CWHumanName[]): string | undefined {
+  if (typeof name === "string") {
+    return name;
+  } else if (Array.isArray(name)) {
+    if (name.length) {
+      const names: string[] = [];
+      name.forEach(n => {
+        if (typeof n === "string") {
+          names.push(n);
+        } else if (typeof n === "object") {
+          const humanNames = getHumanNamesFromObject(n).map(n => n.family);
+          if (humanNames) names.push(humanNames.join(" "));
+        }
+      });
+      return names.join(" ");
+    }
+  } else if (typeof name === "object") {
+    return getHumanNamesFromObject(name)
+      .map(n => n.family)
+      .join(" ");
+  }
+}
+
+/**
+ * Converts a CW name to a FHIR HumanName
+ *
+ * @param name
+ * @returns
+ */
+function convertCWNameToHumanName(
+  name: string | CWHumanName | CWHumanName[]
+): HumanName[] | undefined {
+  if (typeof name === "string") {
+    return [
+      {
+        family: name,
+      },
+    ];
+  } else if (Array.isArray(name)) {
+    if (name.length) {
+      let names: HumanName[] = [];
+      name.forEach(n => {
+        if (typeof n === "string") {
+          names.push({
+            family: n,
+          });
+        } else if (typeof n === "object") {
+          names = getHumanNamesFromObject(n);
+        }
+      });
+    }
+  } else if (typeof name === "object") {
+    return getHumanNamesFromObject(name);
+  }
+}
+
+/**
+ * Converts a CW HumanName to a FHIR HumanName
+ *
+ * @param n
+ * @returns
+ */
+function getHumanNamesFromObject(n: CWHumanName): HumanName[] {
+  const names: HumanName[] = [];
+  if (n.family.length) {
+    for (let i = 0; i < n.family.length; i++) {
+      names.push({
+        family: n.family[i],
+        given: n.given && n.given[i] ? [n.given[i]] : undefined,
+        prefix: n.prefix && n.prefix[i] ? [n.prefix[i]] : undefined,
+      });
+    }
```

Comment:
This mapping doesn't look right, CW name's given and family are diff arrays and can have diff lengths... also, we're not adding `suffix` and `text`. And we get a single human name and return an array.

I think it would be easier to do small adjustment(s) to match CW's HumanName to FHIR's, like CW's having `"unspecified"` `use` that could be translated to `undefined`.

Reply:
I have not encountered a docRef where `.length` of family, given or prefix was more than 1. I assumed that was for when multiple practitioners were involved and recorded together for some reason, i.e.
`family: ["Leite", "Garipov"]`
`given: ["Rafael", "Ramil"]`, 

I thought I'd handle it as such, by creating separate HumanName objects from each of those. 

FHIR Practitioner name is of type `HumanName[]`, and I am mapping it as such. 

Not entirely sure what you mean with the `use` parameter. I have a very vague understanding of how that might be used, but in a few payloads I've seen, it did not come up for CW HumanNames. Would I just return `HumanName[] = []` if `name.use === "unspecified"` ?

Reply:
> I have not encountered a docRef where .length of family, given or prefix was more than 1.

That doesn't mean there can't be doc refs like this. We should assume the CW SDK schemas contain possible combinations of values.

Also, the example you shared:

> family: ["Leite", "Garipov"]
> given: ["Rafael", "Ramil"],

..would come as multiple Practitioners. I don't have a good example where `name` can be an array, but if it does, it would be something like this, I think:

```
"name": [{
  "family": ["Garipov"],
  "given": ["Ramil"]
},{
  "family": ["Leite"],
  "given": ["Rafael"]
}]
```

> Not entirely sure what you mean with the use parameter.

Compre [CW SDK's schema](https://vscode.dev/github/metriport/metriport/blob/develop/packages/commonwell-sdk/src/models/human-name.ts#L28) w/ MedplumFHIR's

What about this?

> I think it would be easier to do small adjustment(s) to match CW's HumanName to FHIR's, like CW's having "unspecified" use that could be translated to undefined.

Reply:
Also, I get a compilation error on local with this branch checked out:

<img width="509" alt="image" src="https://github.com/metriport/metriport/assets/2132564/2e4d5788-4b9d-4230-b565-2f8c1bd97acc">

<img width="901" alt="image" src="https://github.com/metriport/metriport/assets/2132564/58d9be51-e5ad-4e21-a4b9-779073420fa2">